bindings/java/src/main/com/apple/foundationdb/tuple/ByteArrayUtil.java

```diff
@@ -418,5 +418,94 @@ public static String printable(byte[] val) {
 		return s.toString();
 	}
 
+	/**
+	 * Converts the appropriate bytes to Object.
+	 *
+	 * @param bytes The encoded elements.
+	 * @param num Converts the element at {@code num} to Object.
+	 *
+	 * @return The decoded Object.
+	 *
+	 * @throws RuntimeException if an element isn't completely skipped.
```

Comment:
Suppose an element encoded into 5 bytes, if the skip parameter is 3 (how many bytes to skip, the starting byte will be the 4th), it will throw an exception.

Reply:
Okay, I see. I'll say two things (1) this overload of `getObject` doesn't take a `skip` parameter, and (2) I think this might be easier to understand if it's something like "a valid tuple encoded object does not begin at the specified offset".

Also, does this ever actually get thrown? Won't this show up as an `IllegalArgumentException` with some error message like "unsupported tuple type" (in the best case) or garbage (in the worst case)? For example, `\x16\x01\x15\x00` is `(277, null)` packed as a tuple, but if you accidentally provide 1 as the skip instead of 3, then you'll get back `(\x15,)`, right?

Reply:
Hmmm, you are right. If we pass incorrect skip(offset) parameter it can either throw an exception or return garbage.
In TupleUtil.decode function the first byte (offset) is considered as a code (type), if the given offset matches to one of the codes it will try to decode to that type (in this case it will return a garbage or will crash), if there is no matching type it will throw an IllegalArgumentException.